steps/dns/steps.go

```diff
@@ -64,7 +69,39 @@ func (s Steps) InitializeSteps(ctx context.Context, scenCtx *godog.ScenarioConte
 		if !ok {
 			return fmt.Errorf("invalid qtype '%s': permitted values '%s'", qtype, reflect.ValueOf(QueryTypes).MapKeys())
 		}
-		return session.SendQuery(ctx, qt, qname, recursive)
+		switch session.Transport {
+		case "UDP":
+			return session.SendQuery(ctx, qt, qname, recursive)
+		case "DoH with GET":
+			return session.SendDoHQuery(ctx, "GET", qt, qname, recursive)
+		case "DoH with POST":
+			return session.SendDoHQuery(ctx, "POST", qt, qname, recursive)
+		case "DoT":
+			return session.SendDoTQuery(ctx, qt, qname, recursive)
+		default:
+			return fmt.Errorf("unsupported transport protocol. %s", session.Transport)
+		}
+	})
+	scenCtx.Step(`^I send a DoH "([^"]*)" request of type "([^"]*)" for "([^"]*)"(\s\bwithout recursion\b)?$`, func(method, qtype, qname, recursion string) error {
+		recursive := recursion == ""
+		method = golium.ValueAsString(ctx, method)
+		qtype = golium.ValueAsString(ctx, qtype)
+		qname = golium.ValueAsString(ctx, qname)
+		qt, ok := QueryTypes[qtype]
+		if !ok {
+			return fmt.Errorf("Invalid qtype: %s. Permitted values: %s", qtype, reflect.ValueOf(QueryTypes).MapKeys())
+		}
+		return session.SendDoHQuery(ctx, method, qt, qname, recursive)
+	})
+	scenCtx.Step(`^I send a DoT query of type "([^"]*)" for "([^"]*)"(\s\bwithout recursion\b)?$`, func(qtype, qname, recursion string) error {
+		recursive := recursion == ""
+		qtype = golium.ValueAsString(ctx, qtype)
+		qname = golium.ValueAsString(ctx, qname)
+		qt, ok := QueryTypes[qtype]
+		if !ok {
+			return fmt.Errorf("Invalid qtype: %s. Permitted values: %s", qtype, reflect.ValueOf(QueryTypes).MapKeys())
+		}
+		return session.SendDoTQuery(ctx, qt, qname, recursive)
 	})
```

Comment:
It would be interesting to check `session.transport` to avoid problems with steps mixing (between transport and steps definition)

Reply:
These steps are deprecated. Removed in the next commit